src/main/java/jcifs/smb1/http/NtlmHttpURLConnection.java

```
@@ -0,0 +1,631 @@
+        while (entries.hasNext()) {
+            Map.Entry entry = (Map.Entry) entries.next();
+            map.put(entry.getKey(),
+                    Collections.unmodifiableList((List) entry.getValue()));
+        }
+        return Collections.unmodifiableMap(map);
+    }
+
+    public void setInstanceFollowRedirects(boolean instanceFollowRedirects) {
+        connection.setInstanceFollowRedirects(instanceFollowRedirects);
+    }
+
+    public boolean getInstanceFollowRedirects() {
+        return connection.getInstanceFollowRedirects();
+    }
+
+    public void setRequestMethod(String requestMethod)
+            throws ProtocolException {
+        connection.setRequestMethod(requestMethod);
+        this.method = requestMethod;
+    }
+
+    public String getRequestMethod() {
+        return connection.getRequestMethod();
+    }
+
+    public int getResponseCode() throws IOException {
+        try {
+            handshake();
+        } catch (IOException ex) { }
+        return connection.getResponseCode();
+    }
+
+    public String getResponseMessage() throws IOException {
+        try {
+            handshake();
+        } catch (IOException ex) { }
+        return connection.getResponseMessage();
+    }
+
+    public void disconnect() {
+        connection.disconnect();
+        handshakeComplete = false;
+        connected = false;
+    }
+
+    public boolean usingProxy() {
+        return connection.usingProxy();
+    }
+
+    public InputStream getErrorStream() {
+        try {
+            handshake();
+        } catch (IOException ex) { }
+        return connection.getErrorStream();
+    }
+
+    private int parseResponseCode() throws IOException {
+        try {
+            String response = connection.getHeaderField(0);
+            int index = response.indexOf(' ');
+            while (response.charAt(index) == ' ') index++;
+            return Integer.parseInt(response.substring(index, index + 3));
+        } catch (Exception ex) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)